plugins/inputs/influxdb_v2_listener/influxdb_v2_listener.go

```diff
@@ -0,0 +1,359 @@
+	h.authFailures = selfstat.Register("influxdb_v2_listener", "auth_failures", tags)
+	h.routes()
+
+	if h.ReadTimeout.Duration < time.Second {
+		h.ReadTimeout.Duration = time.Second * 10
+	}
+	if h.WriteTimeout.Duration < time.Second {
+		h.WriteTimeout.Duration = time.Second * 10
+	}
+
+	return nil
+}
+
+// Start starts the InfluxDB listener service.
+func (h *InfluxDBV2Listener) Start(acc telegraf.Accumulator) error {
+	h.acc = acc
+
+	tlsConf, err := h.ServerConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	h.server = http.Server{
+		Addr:         h.ServiceAddress,
+		Handler:      h,
+		ReadTimeout:  h.ReadTimeout.Duration,
+		WriteTimeout: h.WriteTimeout.Duration,
+		TLSConfig:    tlsConf,
+	}
+
+	var listener net.Listener
+	if tlsConf != nil {
+		listener, err = tls.Listen("tcp", h.ServiceAddress, tlsConf)
+		if err != nil {
+			return err
+		}
+	} else {
+		listener, err = net.Listen("tcp", h.ServiceAddress)
+		if err != nil {
+			return err
+		}
+	}
+	h.listener = listener
+	h.port = listener.Addr().(*net.TCPAddr).Port
+
+	go func() {
+		err = h.server.Serve(h.listener)
+		if err != http.ErrServerClosed {
+			h.Log.Infof("Error serving HTTP on %s", h.ServiceAddress)
+		}
+	}()
+
+	h.startTime = h.timeFunc()
+
+	h.Log.Infof("Started HTTP listener service on %s", h.ServiceAddress)
+
+	return nil
+}
+
+// Stop cleans up all resources
+func (h *InfluxDBV2Listener) Stop() {
+	err := h.server.Shutdown(context.Background())
+	if err != nil {
+		h.Log.Infof("Error shutting down HTTP server: %v", err.Error())
+	}
+}
+
+func (h *InfluxDBV2Listener) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	h.requestsRecv.Incr(1)
+	h.mux.ServeHTTP(res, req)
+	h.requestsServed.Incr(1)
+}
+
+func (h *InfluxDBV2Listener) handleReady() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.readysServed.Incr(1)
+
+		// respond to ready requests
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusOK)
+		b, _ := json.Marshal(map[string]string{
+			"started": h.startTime.Format(time.RFC3339Nano),
+			"status":  "ready",
+			"up":      h.timeFunc().Sub(h.startTime).String()})
+		res.Write(b)
+	}
+}
+
+func (h *InfluxDBV2Listener) handleDefault() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.notFoundsServed.Incr(1)
+		http.NotFound(res, req)
+	}
+}
+
+func (h *InfluxDBV2Listener) handleWrite() http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		defer h.writesServed.Incr(1)
+
+		bucket := req.URL.Query().Get("bucket")
+
+		body := req.Body
+		// Handle gzip request bodies
+		if req.Header.Get("Content-Encoding") == "gzip" {
+			var err error
+			body, err = gzip.NewReader(body)
+			if err != nil {
+				h.Log.Debugf("Error decompressing request body: %v", err.Error())
+				badRequest(res, err.Error())
+				return
+			}
+			defer body.Close()
+		}
+
+		parser := influx.NewStreamParser(body)
```

Comment:
🤔 so maybe I should still enforce some form of max_body_size if I have to read the entire http response body into memory and pass into `influx.Parse(bodyBytes)` https://github.com/influxdata/telegraf/pull/7828#discussion_r453742459

Reply:
I definitely agree we want to match, not pushing back on that, just trying to brainstorm on the best way to approach this since I presume the accumulator isn't transactional, and I'd have to scrap the streaming code (or again store the telegraf metrics in memory and throw them away if we encounter an error):
https://github.com/influxdata/telegraf/blob/af42bc96bb18bc17360fcf262eb94cebf99cda40/plugins/inputs/influxdb_v2_listener/influxdb_v2_listener.go#L277

Reply:
I do think we should keep `max_body_size`, commented over there.  You are right that the accumulator isn't transactional, so we aren't safe in the case of SIGKILL, power loss or the likes.  Anyway, the output plugin is unsyncronized with the inputs, so the data could go out in separate writes.

Still we can do a close approximation that will work under normal operation (outputs up/clean shutdown): after switching to the Parse() function just check and return the error or add all the metrics if no error.

Reply:
Added max_body_size back, implemented all or nothing writes.